docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,306 @@
+        capabilities:
+          add: ["CHOWN"]
+      command:
+        - /bin/bash
+      args:
+        - -c
+        - >
+          chown -R "1000:1000" "/workspace/source" &&
+          chown -R "1000:1000" "/tekton/home"
+      resources:
+        limits:
+          cpu: "10m"
+          memory: "128Mi"
+        request:
+          cpu: "10m"
+          memory: "128Mi"
+    - name: step-detect
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/detector
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -group=/layers/group.toml
+        - -plan=/layers/plan.toml
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+      resources:
+        limits:
+          cpu: "250m"
+          memory: "50Mi"
+        request:
+          cpu: "250m"
+          memory: "50Mi"
+    - name: step-restore
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/restorer
+      args:
+        - -layers=/layers
+        - -cache-dir=/cache
+        - -group=/layers/group.toml
+      volumeMounts:
+        - name: cache-dir
+          mountPath: /cache
+        - name: layers-dir
+          mountPath: /layers
+    - name: step-build
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/builder
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -group=/layers/group.toml
+        - -plan=/layers/plan.toml
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+      resources:
+        limits:
+          cpu: "500m"
+          memory: "1Gi"
+        request:
+          cpu: "500m"
+          memory: "1Gi"
+    - name: step-export
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/exporter
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -cache-dir=/cache
+        - -group=/layers/group.toml
+        - $(build.output.image)
+      volumeMounts:
+        - name: cache-dir
+          mountPath: /cache
+        - name: layers-dir
+          mountPath: /layers
+```
+
+### Risks and Mitigations
+
+Proper documentation needs to be made to communicate the existance of `flavours` for strategies of the same type to users. When it comes to UI, is the decision of the UI team to decide
+how they will expose(_interface_) this flavours to their end-users. Strategy admins have full responsability on the flavours they decide to define, but recommendations of flavours should be
+made for this repository.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
```

Comment:
We don't necessarily need a lot of details but a little information on how 
* Someone would test it
* Our CI tests would look like to validate this for cluster-scoped and namespace-scoped strategies.

Reply:
ok, let me add this.